app/src/main/java/com/example/asiantech/playmusic/service/MusicService.java

```diff
@@ -0,0 +1,323 @@
+package com.example.asiantech.playmusic.service;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.ContentUris;
+import android.content.Intent;
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+import android.net.Uri;
+import android.os.Binder;
+import android.os.IBinder;
+import android.os.PowerManager;
+import android.provider.MediaStore;
+import android.support.v4.app.NotificationCompat;
+import android.util.Log;
+import android.widget.RemoteViews;
+
+import com.example.asiantech.playmusic.MainActivity;
+import com.example.asiantech.playmusic.MainActivity_;
+import com.example.asiantech.playmusic.MusicReceiver;
+import com.example.asiantech.playmusic.R;
+import com.example.asiantech.playmusic.model.Song;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+/**
+ * @author HoaHT
+ */
+public class MusicService extends Service implements MediaPlayer.OnPreparedListener,
+        MediaPlayer.OnErrorListener, MediaPlayer.OnCompletionListener {
+    private MediaPlayer mMediaPlayer;
+    private ArrayList<Song> mPlayList;
+    private int mSongPosition;
+    private IBinder mMusicBinder = new MusicBinder();
+    private static final int NOTIFICATION_ID = 111;
+    private String mMessage = "";
+    private RemoteViews mRemoteViews;
+    private Notification mNotification;
+    private String mListType = "";
+    private boolean mPause;
+    private int mMode;
+
+    public MusicService() {
+    }
+
+    @Override
+    public void onCreate() {
```

Comment:
WARNING: [CallSuperFirst] super should be called at the start of the method
http://pmd.sourceforge.net/pmd-5.2.3/pmd-java/rules/java/android.html#CallSuperFirst